signature-envelope-jws.md

```diff
@@ -0,0 +1,158 @@
+
+## Storage
+
+A JWS based signature will be persisted with a blob `mediaType` of `"application/jose+json"`
+
+Example
+
+```jsonc
+{
+   "artifactType": "application/vnd.cncf.notary.v2.signature",
+    "blobs": [
+        {
+            "mediaType": "application/jose+json",
+            "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+            "size": 32654
+        }
+    ],
+    "subject": {
+        "mediaType": "application/vnd.oci.image.manifest.v1+json",
+        "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+        "size": 16724
+    },
+    "annotations": {
+        "io.cncf.notary.x509.fingerprint.sha256": "[\"B7A69A70992AE4F9FF103EBE04A2C3BA6C777E439253CE36562E6E98375068C3\" \"932EB6F5598435D4EF23F97B0B5ACB515FAE2B8D8FAC046AB813DDC419DD5E89\"]"
+    }
+}
+```
+
+## JWS Payload
+
+The payload in the JWS envelope is set to BASE64URL encoded value of [Notary v2 Payload](./signature-specification.md#payload).
+
+Example of payload before BASE64URL encoding.
+
+```jsonc
+{
+  "subject": {
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+    "size": 16724,
+    "annotations": {
+        "io.wabbit-networks.buildId": "123"  // user defined metadata
+    }
+  }
+}
+```
+
+## Protected Headers
+
+The JWS envelope for Notary v2 uses following headers
+
+- Registered headers `alg`, `cty`, and `crit`
+- Public headers `io.cncf.notary.signingTime`, `io.cncf.notary.expiry`, `io.cncf.notary.clientID`
+
+Example
+
+```jsonc
+{
+    "alg": "PS384",
+    "cty": "application/vnd.cncf.notary.v2.jws.v1",
+    "io.cncf.notary.signingTime": "2022-04-06 07:01:20.52Z",
+    "io.cncf.notary.expiry": "2022-10-06 07:01:20.52Z",
+    "io.cncf.notary.clientID": "notation/1.0.0",
+    "crit":["io.cncf.notary.expiry"]
+}
+```
+
+- **[`alg`](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.1)**(*string*): This REQUIRED header defines which algorithm was used to generate the signature.JWS specification defines `alg` as a required header, that MUST be present and MUST be understood and processed by verifier. This [section](#supported-alg-header-values) lists the Notary v2 allowed subset of `alg` values supported by JWS.
+- **[`cty`](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.10)**(*string*): The REQUIRED header content-type is used to declare the media type of the secured content (the payload). The supported value is `application/vnd.cncf.notary.payload.v1+json`.
+- **`io.cncf.notary.signingTime`**(*string*): This REQUIRED header specifies the time at which the signature was generated. Its value is a RFC 3339 formatted date time.
```

Comment:
Since time-frac is optional, do we recommend its specification?

Reply:
This is a good call out, we can mention that in signature spec, the optional [`time-secfrac` ](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) MUST NOT be used.

https://datatracker.ietf.org/doc/html/rfc3339#section-5.3 (`time-secfrac`  
```
Rarely used options should be made mandatory or omitted
 for the sake of interoperability whenever possible. 
The format defined below includes only one rarely used option:
 fractions of a second. It is expected that this will be used only by applications which 
require strict ordering of date/time stamps or
which have an unusual precision requirement.
```

Reply:
Updated that fractional seconds SHOULD NOT be used.